Smau/development

sidneymau opened the pull request:
- Updated the API to be more straightforward. The code will run on single coadd block (in however many bands) and produce metadetect catalogs
- The catalogs are returned as a dict of arrow Tables retaining all columns and detections. Metadata is appended to the schema to describe the versions of each software as well as a few other details.
- A unit test will check that the code produces the same catalogs as a reference run
- Various files have been cleaned up (removed the input images and the example notebook)

sidneymau committed:
resolve conflicts

sidneymau committed:
resolve conflicts

sidneymau committed:
Merge branch 'smau/development' of github.com:Roman-HLIS-Cosmology-PIT/metadetect-driver into smau/development

sidneymau committed:
improve scm version handling

sidneymau committed:
remove notebook

sidneymau committed:
linting

sidneymau committed:
add main script

sidneymau committed:
update conda env

sidneymau committed:
working on main

sidneymau committed:
update main

sidneymau committed:
pass mosaic as arg

sidneymau committed:
pass input/output dir as args

sidneymau committed:
update script name

sidneymau committed:
add detection as requirement

sidneymau committed:
upd

sidneymau committed:
support config from cli

sidneymau committed:
use image.shape

sidneymau committed:
load yaml correctly

sidneymau committed:
increase sep max active pixel count

sidneymau committed:
add sklearn for making reports

sidneymau committed:
update output

sidneymau committed:
add (ineffective) logging

sidneymau committed:
update scripts

sidneymau committed:
upd

sidneymau committed:
cleanup comment

sidneymau committed:
write pq

sidneymau committed:
configure metadetect runner

sidneymau committed:
keep separate mdet config

sidneymau committed:
upd

sidneymau committed:
don't require structure in metadetect_config

sidneymau committed:
update report script

sidneymau committed:
Merge branch 'smau/development' of github.com:Roman-HLIS-Cosmology-PIT/metadetect-driver into smau/development

sidneymau committed:
update report figs

sidneymau committed:
fix func args

sidneymau committed:
Merge branch 'smau/development' of github.com:Roman-HLIS-Cosmology-PIT/metadetect-driver into smau/development

sidneymau committed:
update metadetect runner kwarg handling

sidneymau committed:
update report path

sidneymau committed:
decouple from mdet implementation

sidneymau committed:
decouple from mdet implementation

sidneymau committed:
Merge branch 'smau/development' of github.com:Roman-HLIS-Cosmology-PIT/metadetect-driver into smau/development

sidneymau committed:
unused import

sidneymau committed:
add rng/seeds

sidneymau committed:
upd

sidneymau committed:
need np for rng

sidneymau committed:
max seed

sidneymau committed:
update report plots

sidneymau committed:
have scripts for block or mosaic level run

sidneymau committed:
use bands from driver config file

sidneymau committed:
allow null kwargs

sidneymau committed:
grab block from args

sidneymau committed:
bug

sidneymau committed:
update scripts and add some early checks

sidneymau committed:
update reports

sidneymau committed:
updates

sidneymau committed:
add skeelton for postprocessing stage

sidneymau committed:
add script

sidneymau committed:
add postprocess script oops

sidneymau committed:
update postprocess script

sidneymau committed:
start adding code to do extinction correction

sidneymau committed:
check primary mosaic

sidneymau committed:
update postprocess script to working draft

sidneymau committed:
updates to keep things running

sidneymau committed:
upd readme

sidneymau committed:
run black

sidneymau committed:
update example default configs

sidneymau commented:
Marking as ready since I've pushed through a run of the IMCOM 5 sq. deg run using the current state of the code

arunkannawadi commented on .github/workflows/test.yaml:
```diff
@@ -0,0 +1,46 @@
+
+      - name: Run pytest
+        shell: bash -l {0}
+        run: pytest --log-cli-level=INFO tests/
```
Let's turn on coverage with `--cov=.` or something similar

sidneymau replied:
Does something else need to be installed for that to work?
```
$ pytest --log-cli-level=INFO --cov=. tests/
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --cov=.
```

arunkannawadi replied:
You will need at least `pytest-cov` and likely `pytest-xdist` as well.

arunkannawadi commented on src/metadetect_driver/scripts/postprocess.py:
```diff
@@ -0,0 +1,281 @@
+
+    end_time = time.time()
+
+    print(f"Finished postprocessing in {end_time - start_time} seconds")
```
I see logging is imported but unused. Let's just log this at say INFO level instead of printing it.

sidneymau replied:
Removed the logging imports, since these functions are only really ever meant to be run as scripts from the command line and I want it to always emit some of these things

arunkannawadi commented on src/metadetect_driver/driver.py:
```diff
@@ -0,0 +1,669 @@
+# from .defaults import (
+#     DRIVER_DEFAULTS,
+#     METADETECT_DEFAULTS
+# )
```
Remove these commented out code?

arunkannawadi commented on src/metadetect_driver/driver.py:
```diff
@@ -0,0 +1,669 @@
+    #     return self.metadetect_config["metacal"].get("types", ngmix.metacal.METACAL_MINIMAL_TYPES)
+
+    def get_bands(self):
+        return [MetadetectDriver.get_band(outimage) for outimage in self.outimages]
```
Let's make this a property instead of a method

sidneymau replied:
Looking back, this method is only really ever used to set `self.bands` is the `__init__`, which I guess is a bad way of me implementing a property :joy:

sidneymau replied:
(This code is, apparently, unused, but I updated to be a property anyways)

arunkannawadi commented on src/metadetect_driver/driver.py:
```diff
@@ -0,0 +1,669 @@
+            else:
+                _wcs = get_imcom_wcs(outimage)
+                # TODO is there a better way to check for WCS consistency?
+                assert wcs.wcs == _wcs.wcs
```
```suggestion
                assert wcs.wcs == _wcs.wcs, "WCSs are not consistent."
```

arunkannawadi replied:
Any reason why we aren't doing this at the `__init__` stage instead of here?

sidneymau replied:
Hmm, good point, the WCS can probably get parsed/validated there

arunkannawadi commented on src/metadetect_driver/driver.py:
```diff
@@ -0,0 +1,669 @@
+                _wcs = get_imcom_wcs(outimage)
+                # TODO is there a better way to check for WCS consistency?
+                assert wcs.wcs == _wcs.wcs
+        logger.debug(f"WCS is {wcs}")
```
The recommendation best practice is to not use f-strings in loggers. The string formatting takes place even if we are not logging at this level if you do it this way. Instead...

sidneymau replied:
Oh that's a fun detail

arunkannawadi commented on src/metadetect_driver/driver.py:
```diff
@@ -0,0 +1,669 @@
+                _wcs = get_imcom_wcs(outimage)
+                # TODO is there a better way to check for WCS consistency?
+                assert wcs.wcs == _wcs.wcs
+        logger.debug(f"WCS is {wcs}")
```
```suggestion
        logger.debug("WCS is %s", wcs)
```

arunkannawadi reviewed (approved):
LGTM overall! Some minor comments

sidneymau committed:
Update src/metadetect_driver/driver.py

Co-authored-by: Arun Kannawadi <arun.kannawadi@duke.edu>

sidneymau committed:
Update src/metadetect_driver/driver.py

Co-authored-by: Arun Kannawadi <arun.kannawadi@duke.edu>

sidneymau committed:
cleanup some comments

sidneymau committed:
update wcs and bands as properties

sidneymau committed:
update script log imports

sidneymau committed:
report code coverage when testing

sidneymau committed:
update pyrpoject optional dependency syntax

sidneymau committed:
update requirements specifications

sidneymau committed:
Update test/pip

sidneymau committed:
use mamba in actions

sidneymau committed:
cleanup

sidneymau committed:
generalize some of the name/field handling of the postprocess script for both esheldon/metadetect and aguinot/metacoadd

sidneymau committed:
set computed ra/dec _after_ grabbing data from original res in case we need to overwrite columns

aguinot committed:
Setup conda environment to deal with dependencies

Setup a conda environment as some dependencies can only be installed via conda (...) and update the github workflow accordingly.

aguinot committed:
Trigger tests on any PR

aguinot committed:
Fix environment.yml

aguinot committed:
Remove python 3.11 as pyIMCOM requires 3.12 + do not cache

aguinot committed:
Fix mamba version + add python 3.14

aguinot committed:
pre-commit

Lint + formating

aguinot committed:
Fix formating

Add basic documentations when it was missing + simple formating

aguinot committed:
pre-commit formating

aguinot committed:
Fix formating

Add AI generated docstring for test functions

aguinot committed:
Temporarily disable pytest as we don't have test data yet

sidneymau committed:
update env bits

sidneymau committed:
update

sidneymau committed:
simplify lint env

sidneymau committed:
bump to 3.12

sidneymau committed:
python version in mamba

sidneymau committed:
Merge pull request #19 from Roman-HLIS-Cosmology-PIT/fix_merge_conflict

Setup conda environment to deal with dependencies

Thanks for getting this done, @aguinot!
